jscomp/core/lam_compile_group.ml

```diff
@@ -263,6 +263,7 @@ let compile  ~filename output_prefix no_export env _sigs
          lambda_exports are pure
          compile each binding with a return value
          This might be wrong in toplevel
+         TODO: add this check as early as possible in the beginning
```

Comment:
when refactoring, we should determine which identifies are exported as early as we can.
check that sometimes the whole lambda expression can be exported as too